### PR TITLE
(#28) Allow user to listen to the passage

### PR DESCRIPTION
**Pull request by jamesdoc**

### CODE REVIEW

File: src/pages/recite-passage/recite-passage.html [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
   <ion-navbar>
     <ion-title>{{reference}}</ion-title>
+
+    <ion-buttons end>
+      <span class="btnPlayPause">
+        <button class="btnPlayPause__icon" ion-button (click)="onAudioToggle()" aria-label="Listen to the passage">
+          <ion-icon name="{{ playPauseIcon }}"></ion-icon>
+        </button>
+
+        <svg height="36" version="1.1" viewBox="0 0 100 100" width="36" xmlns="http://www.w3.org/2000/svg">
```
Comment (jamesdoc) [inline src/pages/recite-passage/recite-passage.html]: Seems a bit ugly to include the `SVG` this way… I wonder if there is a better way with Ionic?

File: src/pages/recite-passage/recite-passage.html [hunk 2]
```diff
@@ -21,41 +31,22 @@
...
-          <ion-icon name="redo"></ion-icon>
-        </button>
-      </span>
-      <span>
-        <button ion-button *ngIf="endOfPassage && nextPassageExists" (click)="onNext()" aria-label="Go to next passage">
-          <ion-icon name="arrow-forward"></ion-icon>
-        </button>
-      </span>
+    <ion-buttons>
+      <button ion-button (click)="onRecite()" *ngIf="speechEnabled">
```
Comment (jamesdoc) [inline src/pages/recite-passage/recite-passage.html]: Missing `aria-label`

File: src/pages/recite-passage/recite-passage.ts [hunk 3]
```diff
@@ -345,6 +343,34 @@ export class RecitePassagePage {
     this.fetchPassage();
   }
 
+  onAudioToggle = () => {
+    if (!this.passageAudio) {
+      const passageUrl = `http://www.esvapi.org/v2/rest/passageQuery?key=TEST&output-format=mp3&passage=${ this.reference.replace(' ', '.')}`
```
Comment (jamesdoc) [inline src/pages/recite-passage/recite-passage.ts]: Should not deploy the test key there!!! This is why we code review.

@johnraftery - is there a way of putting the ESV key into an `env` file?

File: src/pages/recite-passage/recite-passage.ts [hunk 4]
```diff
@@ -345,6 +343,34 @@ export class RecitePassagePage {
...
+      this.passageAudio = new Audio(passageUrl);
+      this.passageAudio.play();
+      this.playPauseIcon = 'pause';
+
+      this.passageAudio.addEventListener('ended', () => {
+        this.playPauseIcon = 'play';
+      }, false);
+
+      this.passageAudio.addEventListener('timeupdate', function() {
+        let progressBar = <HTMLElement>document.querySelector('.progressBar');
```
Comment (jamesdoc) [inline src/pages/recite-passage/recite-passage.ts]: Ugly - the DOM is being searched every time the `timeupdate` event is fired.

**Comment by johnraftery** [comment]
Looks good; would you like to do the honors?

File: src/pages/recite-passage/recite-passage.html (hunk 2 above)
Comment (jamesdoc) [reply src/pages/recite-passage/recite-passage.html]: Fixing this one now.

File: src/pages/recite-passage/recite-passage.ts (hunk 3 above)
Comment (johnraftery) [reply src/pages/recite-passage/recite-passage.ts]: That would be nice, but from a quick Google, it seems not simple.

Comment (jamesdoc) [reply src/pages/recite-passage/recite-passage.ts]: Can't it just be something like `data.json` with `{ 'esvkey': '########' }` in it? Then read that into the global context?

File: src/pages/recite-passage/recite-passage.html (hunk 1 above)
Comment (jamesdoc) [reply src/pages/recite-passage/recite-passage.html]: I'm not going to fix this here.

File: src/pages/recite-passage/recite-passage.ts (hunk 3 above)
Comment (johnraftery) [reply src/pages/recite-passage/recite-passage.ts]: How do you read this into the global context? I didn't find that in the documentation.

Comment (jamesdoc) [reply src/pages/recite-passage/recite-passage.ts]: What do you make of this. Simple implementation, will do for now.
https://github.com/johnraftery/BibleTrainer/pull/31/commits/de8cc87a77141a3e28c7a275f9de953885ba2d3b

**Pull request merged**